### PR TITLE
Make relative paths in the configuration relative to the configuration file

### PR DESCRIPTION
**Pull request by Revisor**

### CODE REVIEW

File: src/Configuration/ConfigDefinition.php [hunk 1]
```diff
@@ -59,8 +59,8 @@ class ConfigDefinition implements ConfigurationInterface
     const PHP_WORKER_NAME = 'name';
 
     /** Default values */
-    const LOG_DIR_DEFAULT = 'var/log';
-    const CACHE_DIR_DEFAULT = 'var/cache/disqontrol';
+    const LOG_DIR_DEFAULT = '/var/log';
+    const CACHE_DIR_DEFAULT = '../cache/disqontrol';
```
Comment (Revisor) [inline src/Configuration/ConfigDefinition.php]: Or would `/tmp/disqontrol` be a better default cache path?

File: src/Configuration/ConfigDefinition.php [hunk 2]
```diff
@@ -59,8 +59,8 @@ class ConfigDefinition implements ConfigurationInterface
     const PHP_WORKER_NAME = 'name';
 
     /** Default values */
-    const LOG_DIR_DEFAULT = 'var/log';
-    const CACHE_DIR_DEFAULT = 'var/cache/disqontrol';
+    const LOG_DIR_DEFAULT = '/var/log';
```
Comment (Revisor) [inline src/Configuration/ConfigDefinition.php]: `/var/log` is not writable for most processes, usually a new subdirectory is created for each new application, so `/var/log/disqontrol`?

File: src/Configuration/ConfigDefinition.php (hunk 1 above)
Comment (elcuro) [inline src/Configuration/ConfigDefinition.php]: I think that the system path `/tmp/disqontrol` will be a better default path. We don't know structure of the "parent" application

File: src/Disqontrol.php [hunk 3]
```diff
@@ -96,9 +94,14 @@ class Disqontrol
      * @var Container
      */
     private $container;
+    
+    /**
+     * @var string Configuration file path
```
Comment (elcuro) [inline src/Disqontrol.php]: Maybe "Configuration file directory"?

File: src/Disqontrol.php [hunk 4]
```diff
@@ -376,23 +380,67 @@ private function getConfigDir()
...
+        return $this->resolveDirFromConfig($logDir);
+    }
+    
+    /**
+     * Resolve a path from the configuration that can be relative to the
+     * configuration file, or absolute
+     * 
+     * @param string $dir
+     * 
+     * @return string $dir
```
Comment (elcuro) [inline src/Disqontrol.php]: Remove `$dir`

**Pull request merged**